code/modules/mob/living/simple_animal/abnormality/he/red_shoes.dm

```diff
@@ -0,0 +1,410 @@
+		qdel(src)
+		return
+	if(H.sanity_lost)
+		qdel(src)
+		return
+	return ..()
+
+/datum/status_effect/red_possess/on_remove()//causes humans to special panic when applied
+	if(!ishuman(owner))
+		return
+	var/mob/living/simple_animal/hostile/abnormality/red_shoes/shoes
+	var/mob/living/carbon/human/H = owner
+	H.adjust_attribute_buff(PRUDENCE_ATTRIBUTE, 50)
+	if((!shoes) in GLOB.mob_living_list)//if red shoes is suppressed or missing
+		return
+	if(H.sanity_lost)
+		QDEL_NULL(owner.ai_controller)
+		H.ai_controller = /datum/ai_controller/insane/murder/red_possess
+		H.InitializeAIController()
+	return ..()
+
+/datum/status_effect/red_possess/tick()//delete the status if sanity is restored or a panic occurs
+	..()
+	var/mob/living/carbon/human/H = owner
+	if((H.sanityhealth == H.maxSanity) && (sanity_damage_applied == TRUE))
+		qdel(src)
+	if(H.sanity_lost)
+		qdel(src)
+
+//***Custom Panic Definiton***
+/datum/ai_controller/insane/murder/red_possess//define AI controller
+	lines_type = /datum/ai_behavior/say_line/insanity_red_possess
+
+/datum/ai_behavior/say_line/insanity_red_possess
+	lines = list(
+				"Where is everyone?",
+				"Guys, look at me! I’ve got such nice shoes on!",
+				"You all need to see how lovely my shoes are!",
+				"They’re much prettier with blood on them."
+				)
+
+/datum/ai_controller/insane/murder/red_possess/SelectBehaviors(delta_time)//Selects red shoes as the target
+	if(blackboard[BB_INSANE_CURRENT_ATTACK_TARGET] != null)
+		return
+	var/mob/living/simple_animal/hostile/abnormality/red_shoes/shoes
+	for(var/mob/living/simple_animal/hostile/abnormality/red_shoes/M in GLOB.mob_living_list)
+		if(!istype(M))
+			continue
+		shoes = M
+	if (!shoes)//No runtimes after suppression, woohoo!
+		return
+	if(shoes.status_flags & GODMODE)//don't get possessed if it's already breaching
+		current_behaviors += GET_AI_BEHAVIOR(/datum/ai_behavior/desire_move)
+		blackboard[BB_INSANE_CURRENT_ATTACK_TARGET] = shoes
+
+/datum/ai_behavior/desire_move//define AI behavior
+	behavior_flags = AI_BEHAVIOR_REQUIRE_MOVEMENT | AI_BEHAVIOR_MOVE_AND_PERFORM
+	var/list/current_path = list()
+
+/datum/ai_behavior/desire_move/perform(delta_time, datum/ai_controller/controller)//Paths the pancicked to red shoes, causes runtimes if it's dead
+	. = ..()
+	var/mob/living/carbon/human/living_pawn = controller.pawn//the panicked
+	if(IS_DEAD_OR_INCAP(living_pawn))//stop if the panicked is dead
+		return
+	var/mob/living/simple_animal/hostile/abnormality/red_shoes/target = controller.blackboard[BB_INSANE_CURRENT_ATTACK_TARGET]
+	if(!istype(target))
+		finish_action(controller, FALSE)
+		return
+	if(!LAZYLEN(current_path))
+		current_path = get_path_to(living_pawn, target, /turf/proc/Distance_cardinal, 0, 80)
+		if(!current_path) // Returned FALSE or null.
+			finish_action(controller, FALSE)
+			return
+	addtimer(CALLBACK(src, .proc/Movement, controller), 1.25 SECONDS, TIMER_UNIQUE)
```

Comment:
This should be adjusted to be relevant to the user's justice, as it is currently way too slow. The person takes a small singular step every little bit and with the size of facilities it's very easy to stop

Reply:
It's been set to 1 second - (justice /10) seconds with a minimum of 0.1 seconds. Not a great formula but it doesn't get much faster from this line